### PR TITLE
feat(provider/ecs): ECS IAM Role caching classes and tests.

### PR DESCRIPTION
**Pull request by dkirillov**
feat(provider/ecs): ECS IAM Role caching classes and tests.

Adds an IamRoleCachingAgent which is responsible for caching ECS task information, along with IamRoleCacheClient.
The client allows providers/services to interface more easily with the cache.
`getIAM` has been added to AmazonClientProvider.
Models for Roles and TrustRelationships have been added to the core model package.

Tests included! They will be rewritten into groovy+spock, in the future.

@cfieber @ajordens @robzienert @BrunoCarrier

### CODE REVIEW

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/Role.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.model;
+
+import java.util.Collection;
+
+public interface Role {
```
Comment (robzienert) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/Role.java]: Seems to me like IAM models belong in `clouddriver-aws`.

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/IamPolicyReader.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    return trustedEntities;
+  }
+
+
+
```
Comment (robzienert) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/IamPolicyReader.java]: Let's clean up the new lines.

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/IamRoleCachingAgent.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE;
+import static com.netflix.spinnaker.clouddriver.ecs.cache.Keys.Namespace.IAM_ROLE;
+
+public class IamRoleCachingAgent implements CachingAgent {
+
+  public static final String DEFAULT_IAM_REGION = "us-east-1";
```
Comment (robzienert) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/IamRoleCachingAgent.java]: This should use AWS SDK's `Regions.DEFAULT_REGION` instead - we wouldn't want the SDK to change default regions and potentially risk something silly happening (even if IAM is global).

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/Role.java (hunk 1 above)
Comment (dkirillov) [reply clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/Role.java]: Done.

Moved the associated `RoleProvider` and `TrustRelationship` as well.

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/IamPolicyReader.java (hunk 2 above)
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/IamPolicyReader.java]: Done.

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/IamRoleCachingAgent.java (hunk 3 above)
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/IamRoleCachingAgent.java]: Done.

**Comment by robzienert** [comment]
Looks like this branch has some failures before I can merge this.

**Comment by dkirillov** [comment]
@robzienert Do you mean when you run it? And if so can I get more details - exception that's thrown.

Or the Travis check?
https://travis-ci.org/spinnaker/clouddriver/jobs/309129974#L476-L489
I may need you to point me in the right direction if it's the Travis one...

**Comment by robzienert** [comment]
Looks like it just needed a rebase. Everything is good now.

**Pull request merged**